### DISCUSSION TITLE
Style Selector Precedence

### DISCUSSION THREAD

**Discussion by DarkMortainius** [AvaloniaUI/Avalonia#15937] [Q&A]
I have a TreeDataGrid where I have two styles that are conflicting with each other. The problem is the wrong one is taking precedence, and I'm not sure how to get around it.

```XAML
<Style Selector="TreeDataGridRow" x:DataType="registerTable:DataRow">
    <!--Highlight the row if certain flags in the data are set-->
    <Setter Property="Background" Value="{Binding DataPoints, Converter={x:Static converters:Converters.AttributeColorConverter}}"/>
</Style>         
<Style Selector="TreeDataGrid TreeDataGridRow:nth-child(even)">
    <!--Set background to gray on every even row, to make it easier to read-->
    <Setter Property="Background" Value="{DynamicResource AppBackgroundBrush}"/>
</Style>
```

I would like the first style to always have precedence, since it is essentially highlighting any errors in the data, whereas the second style is purely for aesthetics to make it easier to read. I realize the nth-child selector has StyleTrigger precedence, which is why it is always overriding the first style. Is there a way to get around this?

**Comment by stevemonaco** [comment]
The first selector needs to have a conditional activator and needs to be moved after the `nth-child` selector to have precedence.

Some ideas: 1. add an "error" class to `TreeDataGridRow`'s Classes in code when there's an error and adjust the selector. 2. use a property on `TreeDataGridRow` (or an attached property) when there's an error present. ie. `Selector="TreeDataGridRow[Tag=True]"`
